circuits/statetransition/statetransition.go

```diff
@@ -264,10 +287,19 @@ func (circuit StateTransitionCircuit) VerifyMerkleProofs(api frontend.API, hFn u
 // chain of tree transitions of the commitments, and finally the chain of tree
 // transitions of the results. The order of the transitions is fundamental to
 // achieve the final root hash.
-func (circuit StateTransitionCircuit) VerifyMerkleTransitions(api frontend.API, hFn utils.Hasher) {
+func (circuit StateTransitionCircuit) VerifyMerkleTransitions(api frontend.API, hFn utils.Hasher, mask []frontend.Variable) {
 	// verify chain of tree transitions, order here is fundamental.
 	root := circuit.RootHashBefore
 	for i := range circuit.VotesProofs.Ballot {
+		// if the vote is dummy, the transition must be a NOOP (Fnc0=0, Fnc1=0)
+		isDummy := api.Sub(1, mask[i])
+		// Check Ballot proof
+		api.AssertIsEqual(api.Mul(circuit.VotesProofs.Ballot[i].Fnc0, isDummy), 0)
+		api.AssertIsEqual(api.Mul(circuit.VotesProofs.Ballot[i].Fnc1, isDummy), 0)
+		// Check VoteIDs proof
+		api.AssertIsEqual(api.Mul(circuit.VotesProofs.VoteIDs[i].Fnc0, isDummy), 0)
+		api.AssertIsEqual(api.Mul(circuit.VotesProofs.VoteIDs[i].Fnc1, isDummy), 0)
```

Comment:
i'd suggest to improve readability / auditability with a simple helper in MerkleTransition struct, 

in `circuits/merkleproof/merkleproof.go`
```
// AssertDummyIsNoop fails when isDummy is 1 and mp is not a NOOP
func (mp *MerkleTransition) AssertDummyIsNoop(api frontend.API, isDummy frontend.Variable) {
	api.AssertIsEqual(api.Select(isDummy, mp.IsNoop(api), 1), 1)
}
```

something that would allow us to express this block in VerifyMerkleTransitions as simply:

```
		circuit.VotesProofs.Ballot[i].AssertDummyIsNoop(api, isDummy)
		circuit.VotesProofs.VoteIDs[i].AssertDummyIsNoop(api, isDummy)
```